### PR TITLE
fix: escape regular expressions

### LINKED ISSUE #77: 🤔 [ISSUE] - Regular expression injection
### Description

Constructing regular expressions from unvalidated user input can pose a significant security risk. Malicious users may inject carefully crafted patterns into these expressions, causing them to behave unexpectedly and consume excessive computational resources. This can lead to a Denial of Service (DoS) attack, rendering the application unresponsive. To mitigate this threat, it's crucial to sanitize and validate user-provided input before incorporating it into regular expressions.

Weaknesses:
:warning: [CWE-20](https://cwe.mitre.org/data/definitions/20.html)
:warning: [CWE-400](https://cwe.mitre.org/data/definitions/400.html)
:warning: [CWE-730](https://cwe.mitre.org/data/definitions/730.html)

### Reproduction steps

```bash
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
```

### Logs

_No response_

### Browsers

Chrome

### OS

Linux

### PR DESCRIPTION
**Pull request by yassinedorbozgithub**
# Motivation
Constructing regular expressions from unvalidated user input can pose a significant security risk. Malicious users may inject carefully crafted patterns into these expressions, causing them to behave unexpectedly and consume excessive computational resources. This can lead to a Denial of Service (DoS) attack, rendering the application unresponsive. To mitigate this threat, it's crucial to sanitize and validate user-provided input before incorporating it into regular expressions.

Fixes #77

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (Storybook)
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/src/utils/pipes/search-filter.pipe.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ import {
   ArgumentMetadata,
   Logger,
 } from '@nestjs/common';
+import _ from 'lodash';
```
Comment (marrouchi) [inline api/src/utils/pipes/search-filter.pipe.ts:16]:
```suggestion
import escapeRegExp from 'lodash/escapeRegExp';
```
I think this way is better

File: api/migrations/config/create.ts [hunk 2]
```diff
@@ -11,6 +11,8 @@
 import fs from 'fs';
 import path from 'path';
 
+import _ from 'lodash';
```
Comment (marrouchi) [inline api/migrations/config/create.ts:14]: Check my next comment 🙏

**Review by marrouchi — CHANGES_REQUESTED** [review]
Thanks @yassinedorbozgithub

File: api/src/utils/pipes/search-filter.pipe.ts (hunk 1 above)
Comment (yassinedorbozgithub) [reply api/src/utils/pipes/search-filter.pipe.ts:16]: You are right @marrouchi the recommended way will use less size :rocket: 
image
image

**Review by marrouchi — APPROVED** [review]

**Pull request merged**